zerver/views/registration.py

```diff
@@ -137,7 +137,7 @@ def accounts_register(request):
             form = RegistrationForm(
                 initial={'full_name': hesiod_name if "@" not in hesiod_name else ""})
             name_validated = True
-        elif settings.POPULATE_PROFILE_VIA_LDAP:
+        elif settings.POPULATE_PROFILE_VIA_LDAP and not realm_creation:
```

Comment:
The issue is that ToS field in the form is a required field. The form will have the ToS field when `settings.TERMS_OF_SERVICE = True` otherwise it will not have it. When ToS field is present, we won't get into this problem because form won't be valid. When ToS field is not present our form will become valid because all fields are optional.